hyperjumptech/monika
# Adds Redis Monitoring

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Closing issue #858 
2. Add new probe for redis

## How did you implement / how did you fix it  
1.  Using redis lib
2. Adding redis as a "request type"
3. Add to documentation

## How to test  
1. Use the following config `monika.redis.yaml`:
```yaml
probes:
  - id: checking-redis
    name: localhost all works
    alerts:
      - query: response.status == 500
        message: response status message
    incidentThreshold: 1
    recoveryThreshold: 1
    redis:
      - host: 172.17.0.1
        port: 6379
      - host: 172.17.0.2
        port: 6379

  - id: checking-bad-redis
    name: bad redis check
    alerts:
      - query: response.status == 500
        message: response status message
    incidentThreshold: 1
    recoveryThreshold: 1
    redis:
      - host: 172.17.0.10
        port: 6379
        password: mypassword

notifications:
  - id: desktop-01
    type: desktop
```

2. Run your monika with `-c monika.redis.yaml`
4. You should see something like:

![image](https://user-images.githubusercontent.com/964106/188082387-05a1bc67-4b59-4b0d-a7f5-a05ab460fdaa.png)

## Screenshot

Notification
![image](https://user-images.githubusercontent.com/964106/188122864-79ee13f1-6607-472c-9c8d-b7d7e00a29b5.png)

Alert and Notification
![image](https://user-images.githubusercontent.com/964106/188123347-a6522f66-703a-48d4-979d-278f1dd14256.png)

Use of password with and without if Redis has AUTH set:
![image](https://user-images.githubusercontent.com/964106/189575789-2bc4953f-53a2-44f1-b08b-62565ceb152a.png)
